TKGame/Entity.cs

```diff
@@ -105,7 +155,55 @@ public Vector2 Size
         /// <param name="spriteBatch"></param>
         public virtual void Draw(SpriteBatch spriteBatch)
         {
-            spriteBatch.Draw(entityTexture, Position, null, color, 0, Size / 2f, 1f, Orientation, 0);
+            spriteBatch.Draw(UpdateTexture(), Position, null, color, 0, Size / 2f, 1f, Orientation, 0);
+        }
+
+        // Texture Cache to store the final texture of the entity with all the status effects applied to it.
+        // Used to reduce the number of times the entity's texture is updated instead of exchanged.
+        private Dictionary<HashSet<Texture2D>, Texture2D> textureCache = new Dictionary<HashSet<Texture2D>, Texture2D>();
+
+        /// <summary>
+        /// Updates the texture of the entity.
+        /// Stores unique textures in a HashSet and checks if the HashSet already exists in the textureCache.
+        /// If it does not, Combine the textures of the entity with each new unique status effect texture and store it in the textureCache.
+        /// </summary>
+        /// <returns></returns>
+        private Texture2D UpdateTexture()
+        {
+            Texture2D finalTexture = entityTexture;
+
+            if (this.entityType == EntityType.Item ||
+                this.entityType == EntityType.PowerUp ||
+                this.entityType == EntityType.Projectile)
+            {
+                return finalTexture;
+            }
+
+            var statusEffectComponents = this.components.Values.SelectMany(x => x).OfType<IStatusComponent>();
+            HashSet<Texture2D> uniqueTextures = new HashSet<Texture2D>();
+
+            foreach (var statusEffect in statusEffectComponents)
+            {
+                if (statusEffect != null)
+                {
+                    uniqueTextures.Add(statusEffect.GetEffectTexture());
+                }
+            }
+
+            if (textureCache.ContainsKey(uniqueTextures))
+            {
+                finalTexture = textureCache[uniqueTextures];
+            }
+            else
+            {
+                foreach (var texture in uniqueTextures)
+                {
+                    finalTexture = Art.CombineTextures(finalTexture, texture);
+                }
+                textureCache[uniqueTextures] = finalTexture;
+            }
+
+            return finalTexture;
         }
```

Comment:
Maybe move to a high-level entity graphics updating class?

Reply:
I'm not sure how much more complex we want to make our class tree with something like that. If Entity starts inheriting things I can see a lot going wrong. Entity itself as a class isn't an Entity but a generic game object class. My thinking was that this should be in entity since each entity should know how to update it's own texture given it's current status.